### PR TITLE
Duplicate PR CI with github actions

### PR DESCRIPTION
**Pull request by tsipa**
https://github.com/libbpf/libbpf/issues/316

This change supposed to have backward compatibility with travis and with this change i'm expecting that we will run two CIs at the same time.

I would like to get some assistance on coverity - it would be nice if someone can go though https://docs.github.com/en/actions/reference/encrypted-secrets this and set up token

However i can foresee security implications of this - i don't understand what would prevent users from making a PR which would change workflow to print out security token.

### CODE REVIEW

File: .github/actions/vmtest/action.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    description: 'kernel or LATEST'
+    required: true
+    default: 'LATEST'
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        ls -la /tmp/
+        cat /tmp/ci_setup
+        . /tmp/ci_setup
```
Comment (anakryiko) [inline .github/actions/vmtest/action.yml:14]: nit: '.' is easy to miss, `source /tmp/ci_setup` would be a bit more obvious

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+name: bpfci
```
Comment (anakryiko) [inline .github/workflows/test.yml:1]: libbpf-ci?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            cp /tmp/ci_setup $GITHUB_WORKSPACE
+          dockerRunArgs: |
+              --volume "${GITHUB_WORKSPACE}:${GITHUB_WORKSPACE}"
+          shell: /bin/bash
+          install: |
+              export DEBIAN_FRONTEND=noninteractive
+              export TZ="America/Los_Angeles"
+              apt-get update -y
+              apt-get install -y tzdata
+              apt-get install -y build-essential sudo
```
Comment (anakryiko) [inline .github/workflows/test.yml:85]: nit: can combine into a single command?

File: travis-ci/vmtest/build_pahole.sh [hunk 4]
```diff
@@ -15,7 +15,8 @@ cd ${REPO_PATH}
 git init
 git remote add origin ${PAHOLE_ORIGIN}
 git fetch origin
-git checkout master
+# git checkout master
+git checkout 743f2536d8b876a4d9addc6d6d9eb1aa086a47c8
```
Comment (anakryiko) [inline travis-ci/vmtest/build_pahole.sh:19]: pahole got fixed, so we don't need this anymore

File: travis-ci/vmtest/run.sh [hunk 5]
```diff
@@ -436,11 +436,17 @@ sudo umount "$mnt"
...
+if kvm-ok ; then
+  qemu-system-x86_64 -nodefaults -display none -serial mon:stdio \
+    -cpu kvm64 -enable-kvm -smp "$(nproc)" -m 4G \
+    -drive file="$IMG",format=raw,index=1,media=disk,if=virtio,cache=none \
+    -kernel "$vmlinuz" -append "root=/dev/vda rw console=ttyS0,115200$APPEND"
+else
+  qemu-system-x86_64 -nodefaults -display none -serial mon:stdio \
+	  -cpu qemu64 -machine accel=tcg -smp "$(nproc)" -m 4G \
+  	-drive file="$IMG",format=raw,index=1,media=disk,if=virtio,cache=none \
+	  -kernel "$vmlinuz" -append "root=/dev/vda rw console=ttyS0,115200$APPEND"
```
Comment (anakryiko) [inline travis-ci/vmtest/run.sh:448]: Oh, that's a lot of copy/paste of a very long list of arguments... Why not extract what differs into a variable. Something along these lines:

```
if kvm-ok then
QEMU_CPU_ARGS=-cpu kvm64 -enable-kvm
else
QEMU_CPU_ARGS=-cpu qemu64 -machine accel=tcg
fi
qemu-system-x86_64 .... $(QEMU_CPU_ARGS) ....
```
?

**Review by anakryiko — COMMENTED** [review]
Super excited to finally have an alternative to Travis CI!

I don't see any Github Actions runs on this PR. Is there any way to check that before landing PR?

As for the secrets, I can set up what's necessary. Do you know what exactly needs to be done or were you asking for someone to figure it out? But seems like we just hard-code some token in .travis.yaml right now, so if we do the same for Github Actions it won't be a regression :) So we can just start with that, I suppose?

**Comment by tsipa** [comment]
> I don't see any Github Actions runs on this PR. Is there any way to check that before landing PR?
It worked in my repo:
https://github.com/tsipa/libbpf/actions

but i dunno how to trigger CI against this. Maybe that is the answer to my security concern - the changes to CI not reflected on PR.

> As for the secrets, I can set up what's necessary. Do you know what exactly needs to be done or were you asking for someone to figure it out?
I need to have secure variable to be created in this repo and i think i will be able to figure out the rest.

>But seems like we just hard-code some token in .travis.yaml right now, so if we do the same for Github Actions it won't be a regression :) So we can just start with that, I suppose?
I don't know the right answer to this question. If i understand correctly if i do changes to CI code e.g. add echo $SECURE_VAR and send PR with this i would be able to expose secure variable, but mb i don't understand something.

**Comment by anakryiko** [comment]
Great, thanks a lot for working on this.

**Comment by evverx** [comment]
FWIW to judge from https://github.com/evverx/libbpf-fork/pull/2 (where the code should have failed to compile due a missing semicolon) the "vmtest" workflow doesn't test the code on GitHub so it would probably make sense to not run it on PRs and commits and just keep the "cron" job to make sure it's safe to sync the libbpf repository with the kernel tree I assume.

**Pull request merged**